### ISSUE TITLE
Extend the waitFor api to wait for elements

### ISSUE THREAD

**Issue by zabil** [getgauge/taiko#531]
### Current behaviour

Taiko's page actions like `click`, `write` etc waits for the page navigation to complete. This may slow down tests. 

Also in some cases tests need to wait for an action to complete before proceeding. 

### Solution

Extend the `waitFor` API to wait for specific elements to appear on the page so that the user can gracefully handle tests. 

For example speed up tests without loading the entire page. 
```
await click("Checkout", {waitForNavigation: false})
await waitFor("1 item in cart")
await click("Proceed to Pay")
```

or for a process that takes time

```
await click("Place order")
await waitFor("Order Created", intervalSecs(20)) // where creating an order takes 20 seconds
await click("Print")
````

**Comment by dmolin** [comment]
+1

**Comment by supraja-kr** [comment]
We are facing the same challenge in our team too.

**Comment by Debashis9012** [comment]
I have validated the above feature request Kindly find the below observation:

 I have used the ``` waitFor("element",5000)``` and ``` waitFor("element")``` to load the element properly so that we can perform the actions(like click/enter text) gracefully.

It seems its only checking for the text and once it finds out the text on the screen then the control is moving to next line to perform the action on the webelement. 
But in real the element is taking more time to gets loaded(or taking more time to load in a stable condition) on the screen which leads to fail perform correct action. Until the element loads properly we cant perform any correct action on it.
Kindly find the below script for reference:
```
const { openBrowser, goto, text, click, below, link, toRightOf } = require('taiko');

(async () => {
    try {
        await openBrowser({headless:false});
        await goto("www.zomato.com/bangalore");
        await click(text("Pocket-Friendly Delivery"));
        //await waitFor("Order Food Online",5000)
        await waitFor("Order Food Online")
        await click("Order Food Online");
    } catch (e) {
        console.error(e);
    } finally {
    //    closeBrowser();
    }
})();
```
Commit: ```082018c0c4081144b2dc19b0f375d0a256d94b3c```

**Comment by Apoorva-GA** [comment]
The problem is that, in the above script we should wait for the new element that is shifting the layout to exist, which is same as [issue](https://github.com/getgauge/taiko/issues/558).

Also the fix to waitFor to element selected using a selector is fixed as part of a84409d.

**Comment by Debashis9012** [comment]
This issue has been verified and found fixed with the commit version ``` a84409d ```
